src/routes/transactions/transactions-history.controller.spec.ts

```diff
@@ -804,6 +804,200 @@ describe('Transactions History Controller (Unit)', () => {
+    const chain = chainBuilder().build();
+    const untrustedToken = tokenBuilder().with('trusted', false).build();
+    // Use same date so that groups are created deterministically
+    const date = faker.date.recent();
+    const oneDayAfter = new Date(date.getTime() + 1000 * 60 * 60 * 24);
+    const transactionHistoryData = {
+      count: faker.number.int(),
+      next: faker.internet.url(),
+      previous: faker.internet.url(),
+      results: [
+        ethereumTransactionToJson(
+          ethereumTransactionBuilder()
+            .with('transfers', [
+              erc20TransferToJson(
+                erc20TransferBuilder()
+                  .with('tokenAddress', untrustedToken.address)
+                  .build(),
+              ) as Transfer,
+              erc20TransferToJson(
+                erc20TransferBuilder()
+                  .with('tokenAddress', untrustedToken.address)
+                  .build(),
+              ) as Transfer,
+            ])
+            .with('executionDate', date)
+            .build(),
+        ),
+        ethereumTransactionToJson(
+          ethereumTransactionBuilder()
+            .with('transfers', [
+              erc20TransferToJson(
+                erc20TransferBuilder()
+                  .with('tokenAddress', untrustedToken.address)
+                  .build(),
+              ) as Transfer,
+              erc20TransferToJson(
+                erc20TransferBuilder()
+                  .with('tokenAddress', untrustedToken.address)
+                  .build(),
+              ) as Transfer,
+            ])
+            .with('executionDate', oneDayAfter)
+            .build(),
+        ),
+      ],
+    };
+    networkService.get.mockImplementation((url) => {
+      const getChainUrl = `${safeConfigUrl}/api/v1/chains/${chain.chainId}`;
+      const getAllTransactions = `${chain.transactionService}/api/v1/safes/${safe.address}/all-transactions/`;
+      const getSafeUrl = `${chain.transactionService}/api/v1/safes/${safe.address}`;
+      switch (url) {
+        case getChainUrl:
+          return Promise.resolve({ data: chain });
+        case getAllTransactions:
+          return Promise.resolve({ data: transactionHistoryData });
+        case getSafeUrl:
+          return Promise.resolve({ data: safe });
+        case `${chain.transactionService}/api/v1/tokens/${untrustedToken.address}`:
+          return Promise.resolve({ data: untrustedToken });
+        default:
+          return Promise.reject(new Error(`Could not match ${url}`));
+      }
+    });
+
+    await request(app.getHttpServer())
+      .get(
+        `/v1/chains/${chain.chainId}/safes/${safe.address}/transactions/history`,
+      )
+      .expect(200)
+      .expect((response) => {
+        // Empty array
+        expect(response.body['results']).toEqual([]);
+      });
+  });
+
+  it('Should avoid returning a date label if all the token transfers for that date are untrusted', async () => {
```

Comment:
"Should not return".

Avoid implies that there are some indeterministic factor to it.

Reply:
Thanks! I agree, changed in [5fcf5d3](https://github.com/safe-global/safe-client-gateway/pull/983/commits/5fcf5d306277a6d1aee63d6b13777062a03879c2)